bench/wtperf/wtperf.c

```diff
@@ -481,11 +482,27 @@ worker(void *arg)
         if (opts->index_like_table)
             generate_index_key(thread, false, index_buf, next_val);
 
-        if (workload->table_index == INT32_MAX)
+        if (workload->table_index == INT32_MAX) {
```

Comment:
At the top of this function (`worker()`) there is a call to:
```
if ((ret = session->open_cursor(
               session, wtperf->uris[workload->table_index], NULL, NULL, &cursor)) != 0) {
```
that should not take place if `reopen_cursor` is true.